src/api/app/models/kiwi/image/xml_builder.rb

```diff
@@ -13,11 +13,37 @@ def build
 
         doc = update_packages(doc)
         doc = update_repositories(doc)
+        doc = update_description(doc)
+
         Nokogiri::XML(doc.to_xml, &:noblanks).to_xml(indent: kiwi_indentation(doc))
       end
 
       private
 
+      def update_description(document)
+        return document if @image.description.blank?
+
+        if document.xpath('image/description[@type="system"]').first
+          %w[author contact specification].each do |element_name|
+            update_description_element(document, element_name)
+          end
+        else
+          document.at_css('image').first_element_child.before(@image.description.to_xml)
+        end
+        document
+      end
+
+      def update_description_element(document, element_name)
+        element_xml = document.xpath("image/description[@type='system']/#{element_name}").first
+        unless element_xml
+          element_xml = Nokogiri::XML::Node.new(element_name, document)
+          document.xpath('image/description[@type="system"]').first.add_child(element_xml)
+        end
+        # rubocop:disable Lint/UselessSetterCall
```

Comment:
Why don't you do this in 2 steps, first calculate the element_xml in one method and then update the content outside..

Reply:
Good idea, fixed.